packages/synapse-react-client/src/components/EntityHeaderTable/DebouncedInput.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react'
+import { TextField, TextFieldProps, Autocomplete } from '@mui/material'
+import { useDebouncedEffect } from '../../utils/hooks'
+
+// A debounced input react component
+
+export function DebouncedInput({
+  value: initialValue,
```

Comment:
Rename prop to `initialValue` because the internal `value` will not update if this prop updates...if you want this to act as 'fully-controlled', and have the value update when the prop updates, I think you may need to debounce where you apply the filter, instead of where you fire the input's `onChange`.